src/OptFrame/Heuristics/EvolutionaryAlgorithms/PSO.hpp

```diff
@@ -264,6 +113,22 @@ class PSO : public SingleObjSearch<XES, XES2, XSH2>
         }
         return swarm;
     }
+    
+    void BirdMutation ( Bird b)
+    {
+       if (rand01() < 0.1*0.1)
+       {
+          for (unsigned j = 0; j < this->cS.size(); j++ )
+          {
+            double r = rand01();
+            if (r < 0.5)
+               b.position[j] += (cS[j] - cI[j]) * (pow(2.0 *   r   , 1.0 / 21 ) - 1);
+            else
+               b.position[j] += (cS[j] - cI[j]) * (pow(2.0 * (1-r) , 1.0 / 21 ) + 1);
```

Comment:
better to write `::pow`, so you make sure the global scope `<math.h>` is being used (not some other function).